From 489104265b77e5d9b1d32ae4118acb596caa8594 Mon Sep 17 00:00:00 2001
From: Wen Gong <quic_wgong@quicinc.com>
Date: Thu, 8 Sep 2022 22:24:39 -0400
Subject: [PATCH 100/112] wifi: mac80211: set all links active for station by
 default - 2d37217f46b17cd7bf993e1f2d850a844a1c01b5 wifi: mac80211: isolate
 driver from inactive links

Signed-off-by: Wen Gong <quic_wgong@quicinc.com>
Signed-off-by: Amutha Ravikumar <quic_aravikum@quicinc.com>
---
 net/mac80211/link.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

--- a/net/mac80211/link.c
+++ b/net/mac80211/link.c
@@ -222,17 +222,9 @@ static void ieee80211_set_vif_links_bitm
 
 	switch (sdata->vif.type) {
 	case NL80211_IFTYPE_AP:
-		/* in an AP all links are always active */
-		sdata->vif.active_links = valid_links;
-
-		/* AP links are not expected to be disabled */
-		WARN_ON(dormant_links);
-		break;
 	case NL80211_IFTYPE_STATION:
-		if (sdata->vif.active_links)
-			break;
+		/* in an AP and STA all links are always active */
 		sdata->vif.active_links = valid_links & ~dormant_links;
-		WARN_ON(hweight16(sdata->vif.active_links) > 1);
 		break;
 	default:
 		WARN_ON(1);
